src/sonic-config-engine/minigraph.py

```diff
@@ -1771,17 +1771,16 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
         port_default_speed =  port_speeds_default.get(port_name, None)
         port_png_speed = port_speed_png[port_name]
 
+        ports.setdefault(port_name, {})['speed'] = port_speed_png[port_name]
+
         # when the port speed is changes from 400g to 100g/40g
         # update the port lanes, use the first 4 lanes of the 400G port to support 100G/40G port
         if port_default_speed == '400000' and (port_png_speed == '100000' or port_png_speed == '40000'):
             port_lanes =  ports[port_name].get('lanes', '').split(',')
-            # check if the 400g port has only 8 lanes
-            if len(port_lanes) != 8:
-                continue
```

Comment:
i think this is not the problem, we should keep this.

theoritical speaking, there about 2lane, each lane is 200G, total 400G configuration. in this case, you do not want to select 4 lanes since the port only has 2 lanes.

Reply:
@vdahiya12 
consider following cases:-
1. 400G using 8x50G
2. 400G using 4x100G
3. 400G using 2x200G (in the future)

A. We should skip in the #3 case because we don't have 4 lanes in port_config.ini to choose from for 100G

B. #2 In this case we don't need to select the lanes for 100G speed because its already 4 lanes

So, the only case which can support 100G NRZ is when 400G is 8*50G so we should skip updating the lanes to first 4 lanes if the number of lanes is not 8

Reply:
fixed